### PR TITLE
update caffeloader with the change of parameters

### PR DESCRIPTION
**Pull request by SeaOfOcean**
## What changes were proposed in this pull request?

update caffeloader with the change of parameters

## How was this patch tested?

manual check

### CODE REVIEW

File: dl/src/main/scala/com/intel/analytics/bigdl/utils/CaffeLoader.scala [hunk 1]
```diff
@@ -135,8 +136,8 @@ class CaffeLoader[T: ClassTag](prototxtPath: String, modelPath: String,
         logger.info(s"$name uses initialized parameters")
         return
       }
-      val (weight, _) = loadParameters(name, mod.parameters()._1)
```
Comment (qiuxin2012) [inline dl/src/main/scala/com/intel/analytics/bigdl/utils/CaffeLoader.scala]: You can try getParametersTable, to set get the parameter.

Comment (qiuxin2012) [inline dl/src/main/scala/com/intel/analytics/bigdl/utils/CaffeLoader.scala]: And throw an exception when the parameters don't match with caffe's model.

File: dl/src/main/scala/com/intel/analytics/bigdl/utils/CaffeLoader.scala [hunk 2]
```diff
@@ -75,79 +74,82 @@ class CaffeLoader[T: ClassTag](prototxtPath: String, modelPath: String,
...
-        s"data shape in caffe is ${ caffeWeight.get.getShape() }," +
-        s" while data shape in bigdl is ${ destPara(0).size().mkString(",") }")
-    require(destPara(0).isContiguous())
-    val weightData = destPara(0).storage().array()
-    for (i <- 0 until weightList.size()) {
-      weightData(i) = ev.fromType[Float](weightList.get(i))
+    if (caffeWeight.isDefined) {
+      require(params.contains("weight"), s"$name should contain weight")
+      val caffeWeightData = caffeWeight.get.getDataList
+      val weight = params[Tensor[T]]("weight")
```
Comment (qiuxin2012) [inline dl/src/main/scala/com/intel/analytics/bigdl/utils/CaffeLoader.scala]: how about params don't contain weight?

File: dl/src/main/scala/com/intel/analytics/bigdl/utils/CaffeLoader.scala [hunk 3]
```diff
@@ -75,79 +74,82 @@ class CaffeLoader[T: ClassTag](prototxtPath: String, modelPath: String,
...
-    if (destPara.length > 1 && destPara(1) != null) {
-      val caffeBias = getBlob(name, 1)
-      if (caffeBias.isEmpty) return (destPara(1), null)
-      val biasList = caffeBias.get.getDataList
-      require(destPara(1).nElement() == biasList.size(),
+    val caffeBias = getBlob(name, 1)
+    if (caffeBias.isDefined) {
+      require(params.contains("bias"), s"$name should contain bias")
+      val caffeBiasList = caffeBias.get.getDataList
+      val bias = params[Tensor[T]]("bias")
```
Comment (qiuxin2012) [inline dl/src/main/scala/com/intel/analytics/bigdl/utils/CaffeLoader.scala]: how about params don't contain bias?

File: dl/src/main/scala/com/intel/analytics/bigdl/utils/CaffeLoader.scala (hunk 2 above)
Comment (SeaOfOcean) [inline dl/src/main/scala/com/intel/analytics/bigdl/utils/CaffeLoader.scala]: if caffe layer contains weight, then weight is required in bigdl module, as shown in line 90

File: dl/src/main/scala/com/intel/analytics/bigdl/utils/CaffeLoader.scala (hunk 3 above)
Comment (SeaOfOcean) [inline dl/src/main/scala/com/intel/analytics/bigdl/utils/CaffeLoader.scala]: if caffe layer contains bias, then bias is required in bigdl module, as shown in line 108

**Pull request merged**